django_cloud_deploy/crash_handling/__init__.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A module handles crash of the tool."""
+
+import os
+import platform
+import subprocess
+import sys
+import traceback
+import urllib.parse
+import webbrowser
+
+import jinja2
+
+from django_cloud_deploy.cli import io
+from django_cloud_deploy import __version__
+
+# TODO: Add an issue label for issues reported by users
+_REQUEST_URL_TEMPLATE = (
+    ('https://github.com/GoogleCloudPlatform/django-cloud-deploy/issues/new?'
+     'body={}'))
+
+
+with open(os.path.join(os.path.dirname(__file__), 'template',
+                       'issue_template.txt')) as f:
+    _ISSUE_TEMPLATE = f.read()
+
+
+def handle_crash(err: Exception, command: str,
+                 console: io.IO = io.ConsoleIO()):
+    """The tool's crashing handler.
+
+    Args:
+        err: The exception get thrown.
+        command: The command causing the exception to get thrown.
+        console: Object to use for user I/O.
+    """
+
+    # TODO: Only handle crashes caused by our code, not user's code.
+    console.tell(
+        'django-cloud-deploy crashed with ({}): {}'.format(
+            type(err).__name__, err))
+
+    ans = console.ask(
+        'Would you like to file a bug against Django Cloud Deploy?\n'
+        'If you want, we will open a browser to direct you to the\n'
+        'tool\'s Github repo. The bug will include information about\n'
+        'your environment but you can remove any information that you\n'
```

Comment:
The bug will include information regarding the crash and information that will help us prevent this in the future.